### PR TITLE
Docs: Improve deployment and quickstart docs and tutorials

### LINKED ISSUE #2162: allow for easier usability of the tutorials from their `.html` build within the docs
**Is your feature request related to a problem? Please describe.**
After looking at a cool tutorial, it is a bit unclear how it can be used easily without deep diving into the docs.

**Describe the solution you'd like**
I want to add references for starting argilla
1 Google Colab
2 HuggingFace spaces
3 quickstart/docker
4 cloud offering

I am considering several options
- download button for the corresponding `.ipynb`
- host them on Colab

**Describe alternatives you've considered**
Look in GitHub and download the docs there.

**Additional context**
N.A.

### PR DESCRIPTION
**Pull request by dvsrepo**
# Description

Adds Open in Colab button at the top of each notebook-generated docs page. We still need to review the `Setup` section of each notebooks to add context about new deployment types and normalize everything. This would actually add the button to deep-dives or other notebooks but I think is more a feature than a bug as people can follow through by running code cells.

Closes #2162 

**Type of change**
- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation update

**How Has This Been Tested**

Please describe the tests that you ran to verify your changes. And ideally reference `tests`.

- [x] Tested locally with autobuild sphinx docs

### CODE REVIEW

**Comment by dvsrepo** [comment]
@davidberenstein1957 finally my robot friend was talking shit, it should be as simple as this. We can do the same with a Jupyter badge pointing to the ipynb, and some other crazy ideas that come to mind like linking to a live dataset inside a deployed Argilla instance (might involve mapping dataset names with file names, but why not 🥸 )

One thing to decide:

- The URL for the file should point to develop by default, main?

**Comment by davidberenstein1957** [comment]
@dvsrepo I corrected the Open in Colab Button and added a Source.ipynb button. Also, I restructured the docs section w.r.t. quickstart and deployments. I am not sure if this is clearer, but perhaps you have some more ideas. https://docs.argilla.io/en/docs-open_in_colab_button/

**Comment by davidberenstein1957** [comment]
I want to remove Setup and installation because it is kind of duplicated from quickstart, and just use the "deploy" and "configure"

File: docs/_source/conf.py [hunk 1]
```diff
@@ -143,27 +171,34 @@
...
-    notfound_urls_prefix = None
-else:
-    lang, docs_version = (
-        os.getenv("READTHEDOCS_LANGUAGE", "en"),
-        os.getenv("READTHEDOCS_VERSION", "latest"),
-    )
+.. raw:: html
 
-    notfound_urls_prefix = f"/{lang}/{docs_version}/"
+    <div class="row">
```
Comment (dvsrepo) [inline docs/_source/conf.py]: I'd recommend to have both buttons inline on the same row, is it possible?

File: docs/_source/conf.py [hunk 2]
```diff
@@ -143,27 +171,34 @@
...
+    <div class="col">
+    <a href="https://colab.research.google.com/github/argilla-io/argilla/tree/"""
```
Comment (dvsrepo) [inline docs/_source/conf.py]: I would add `target="_blank"` so the links are opened on a new page

File: docs/_source/conf.py [hunk 3]
```diff
@@ -143,27 +171,34 @@
...
+    + os.getenv("READTHEDOCS_VERSION", "main").replace("-", "/")
+    + """/docs/_source/{{ env.doc2path(env.docname, base=None) }}"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a></div>"""
```
Comment (dvsrepo) [inline docs/_source/conf.py]: For Colab (on the branch rtd deployment) I'm getting the following error. It seems like blob is missing:

```
malformed GitHub path: missing 'blob' before branch name: argilla-io/argilla/tree/docs/open_in_colab_button/docs/_source/tutorials/notebooks/deploying-textclassification-colab-activelearning.ipynb
```

File: docs/_source/conf.py [hunk 4]
```diff
@@ -143,27 +171,34 @@
...
+    + """<div class="col"> <a href="https://github.com/argilla-io/argilla/tree/"""
+    + os.getenv("READTHEDOCS_VERSION", "main").replace("-", "/")
+    + """/docs/_source/{{ env.doc2path(env.docname, base=None) }}"><img src="https://img.shields.io/badge/github-view%20source-black.svg" alt="View Source on GitHub"/></a></div>
```
Comment (dvsrepo) [inline docs/_source/conf.py]: Can we say "View Notebook on GitHub"?

File: docs/_source/_common/next_steps.html [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<section id="Next-steps">
+    <h2>Next steps<a class="headerlink" href="#Next-steps" title="Permalink to this headline">#</a></h2>
+    <p>⚙️ <a class="reference internal" href="#setup-argilla">The section</a> about setting up Argilla and opening or
+        downloading the notebook.</p>
+    <p>📚 <a class="reference external" href="https://docs.argilla.io">Argilla documentation</a> for more guides and
+        tutorials.</p>
+    <p>🙋‍♀️ Join the Argilla community! A good place to start is the <a class="reference external"
```
Comment (dvsrepo) [inline docs/_source/_common/next_steps.html]: I would add the join Slack community paragraph from this tutorial: https://docs.argilla.io/en/latest/tutorials/notebooks/labelling-textclassification-sentence-transformers-semantic.html#Next-steps, before the Argilla documentation paragraph.

Let's discuss briefly about the first paragraph about setup

**Comment by dvsrepo** [comment]
@leiyre we need some quick css position adjustments for two new buttons that will appear on top of tutorials and guides.

For mobile they overlap with slack and stars so we can add more space on top of the buttons (or maybe hide them all together as they're not very useful @davidberenstein1957 what do you think?)

Screenshot_20230120-162957

For desktop, at least some more space and maybe align them to the right? If we align them to the right there's the dark mode button too we need to take into account. I let you decide the most simple, quick solution.

The html mark-up is on the prolog section of the conf.py file.

Here's the link to docs for this branch: https://docs.argilla.io/en/docs-open_in_colab_button/tutorials/notebooks/template.html

Can be done on Monday to finish this issue

**Pull request merged**